src/character.cpp

```diff
@@ -3110,27 +3110,24 @@ void find_ammo_helper( T &src, const item &obj, bool empty, Output out, bool nes
                     return VisitResponse::SKIP;
                 }
 
-                if( node->is_container() && !node->is_container_empty() ) {
-                    return VisitResponse::NEXT;
-                }
-
                 if( node->typeId() == contents_id ) {
                     out = item_location( src, node );
-                    return VisitResponse::ABORT;
```

Comment:
If ABORT is returned then the search ends after finding just one item. Keeping that in place never gives the player a chance to choose which source because it returns only one item and ends.